libs/util/git/src/providers/aws/aws-code-commit.service.ts

```diff
@@ -60,81 +68,185 @@ export class AwsCodeCommitService implements GitProvider {
+        const propValue = value[prop];
+        return propValue !== null && propValue !== undefined;
+      });
+    }
+    return false;
+  }
+
   async init(): Promise<void> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getGitInstallationUrl(amplicationWorkspaceId: string): Promise<string> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getCurrentOAuthUser(accessToken: string): Promise<CurrentUser> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getOAuthTokens(authorizationCode: string): Promise<OAuthTokens> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async refreshAccessToken(): Promise<OAuthTokens> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getGitGroups(): Promise<PaginatedGitGroup> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
+
   async getRepository(
     getRepositoryArgs: GetRepositoryArgs
   ): Promise<RemoteGitRepository> {
-    throw new Error("Method not implemented.");
+    const { repositoryName } = getRepositoryArgs;
+    const command = new GetRepositoryCommand({
+      repositoryName,
+    });
+    const { repositoryMetadata } = await this.awsClient.send(command);
+
+    if (this.isRequiredValid(repositoryMetadata)) {
+      return {
+        admin: false,
+        defaultBranch: repositoryMetadata.defaultBranch,
+        fullName: repositoryMetadata.repositoryName,
+        name: repositoryMetadata.repositoryName,
+        private: true,
+        url: repositoryMetadata.cloneUrlHttp,
+        groupName: null,
+      };
+    } else {
+      throw new Error(`Repository ${repositoryName} not found`);
+    }
   }
+
   async getRepositories(
     getRepositoriesArgs: GetRepositoriesArgs
   ): Promise<RemoteGitRepos> {
-    throw new Error("Method not implemented.");
+    const { pagination } = getRepositoriesArgs;
+    const command = new ListRepositoriesCommand({
+      sortBy: "repositoryName",
+      order: "ascending",
+    });
+
+    // TODO improve pagination
+    const { repositories } = await this.awsClient.send(command);
+    let paginatedRepositories: RemoteGitRepository[] = [];
+    if (repositories && repositories.length > 0) {
+      const skip = (pagination.page - 1) * pagination.perPage;
+      const take = skip + pagination.perPage;
+      paginatedRepositories = repositories
+        .slice(skip, take)
+        .map((repository) => {
+          if (this.isRequiredValid(repository)) {
+            return {
+              admin: false,
+              defaultBranch: "",
+              fullName: repository.repositoryName,
+              name: repository.repositoryName,
+              private: true,
+              url: "",
+              groupName: null,
+            };
+          } else {
+            throw new Error(
```

Comment:
Do you want to throw in this case or skip it and not add it to paginatedRepositories?

Reply:
Not needed since we defined the return type for the function and the compiler will complain if this method returns a different object